File: apps/cli/commands/stop.ts
/**
 * Stop command - Terminate the daemon
 */
import { getDaemonStatus, killDaemon } from "../daemon-manager.js";
import { stopOpencodeServe } from "../opencode-sidecar.js";

export async function stopDaemon(): Promise<void> {
  const status = await getDaemonStatus();

  if (!status.running) {
    console.log("Daemon is not running");
    process.exit(1);
  }

  console.log(`Stopping daemon (PID: ${status.pid})...`);

  try {
    await killDaemon(status.pid!);
    console.log("Daemon stopped successfully");
  } catch (error) {
    console.log(`Failed to stop daemon: ${error}`);
    process.exit(1);
  }

  try {
    if (await stopOpencodeServe()) console.log("OpenCode serve sidecar stopped");
  } catch (error) {
    console.log(`Failed to stop OpenCode serve sidecar: ${error}`);
  }
}

File: apps/server-go/internal/cli/commands/stop.go
// Stop command - terminate the daemon. Port of apps/cli/commands/stop.ts.
package commands

import (
	"fmt"
	"os"

	"github.com/Adelodunpeter25/console/apps/server-go/internal/cli/daemon"
)

// StopDaemon terminates the running daemon.
func StopDaemon() error {
	status := daemon.GetStatus()
	if !status.Running {
		fmt.Println("Daemon is not running")
		os.Exit(1)
	}

	fmt.Printf("Stopping daemon (PID: %d)...\n", status.Pid)
	if err := daemon.KillDaemon(status.Pid); err != nil {
		fmt.Printf("Failed to stop daemon: %v\n", err)
		os.Exit(1)
	}
	fmt.Println("Daemon stopped successfully")
	return nil
}
